core/logger/manager.php

```diff
@@ -75,9 +78,19 @@ public function rest_error_handler( $error_number, $error_message, $error_file,
 			}
 		}
 
-		$this->shutdown( $error->get_error_data() );
+		$error_data = $error->get_error_data();
+
+		if ( ! $skip_shutdown ) {
+			$this->shutdown( $error_data );
+
+			exit;
+		}
+
+		return $error_data;
```

Comment:
Hey, If I understand it correctly it prints the JSON to the client and then returns some error data. maybe we can do something else? seems a bit confusing.

Reply:
The return is used in tests only.